src/Molecules/Header/Header.jsx

```diff
@@ -0,0 +1,94 @@
+import React, { useState, useRef } from 'react'
+import { Link as RouterLink } from 'react-router-dom'
+import { Link, Typography, CircularProgress } from '@material-ui/core'
+import { Edit, Save } from '@styled-icons/material'
+
+import Input from '../../Atoms/Input'
+import { useTranslation } from '../../Atoms/Trans'
+import FlexBox from '../../Templates/FlexBox'
+import BottomTooltipIcon from './BottomTooltipIcon'
+
+const Header = ({
+  children,
+  feature = '',
+  featurePath = null,
+  subFeature = '',
+  onChange = () => {},
+  onSave = false,
+  status = null,
+  actions = null,
+  ...props
+}) => {
+  const saveButton = useRef(null)
+  const [isEditing, setEditing] = useState(false)
+  const [isSaving, setSaving] = useState(false)
+  const [value, setValue] = useState(subFeature)
+
+  const t = useTranslation()
+  const displayMode = !isEditing && !isSaving
+  const editMode = isEditing && !isSaving
+
+  const handleValue = value => setValue(value) || onChange(value)
+  const handleCancel = () => handleValue(subFeature) || setEditing(false) || setSaving(false)
+  const handleSave = () => {
+    setEditing(false)
+    setSaving(true)
+
+    let response = typeof onSave === 'function' ? onSave(value) : onSave
+
+    if (!(response instanceof Promise)) {
+      response = response ? Promise.resolve() : Promise.reject(new Error(JSON.stringify(response)))
+    }
+
+    response.then(() => handleValue(value) || setSaving(false)).catch(handleCancel)
```

Comment:
Don't consider an error as a cancel, the input should be editable again with the submitted value